### PR TITLE
Rename record merging policy to /record/merge

### LINKED ISSUE #4063: Add record merging permission as a policy
> Is it possible to set permissions so that certain users are prevented from using the merging tool, while retaining full access to the agents table?
> 
> As admin, it would be nice to be able to easily control it via the tool dropdown.

The issue is just because someone has permission to read, create, and update all tables **record by record** they shouldn't necessarily have the ability to perform bulk updates across the database.

There should be an independent permission for merging in the tool menu or query builder

image

**Requested By:** Cris at Gothenburg museum of natural history

### PR DESCRIPTION
**Pull request by melton-jason**
Fixes #4063

### CODE REVIEW

**Review by realVinayak — APPROVED** [review]
Tested by making merge policies in previous version of specify, they get correctly renamed!

File: specifyweb/notifications/migrations/0004_rename_merge_policy.py
Comment (maxpatiiuk) [inline specifyweb/notifications/migrations/0004_rename_merge_policy.py:1]: 👏 great job for not forgetting to do this

**Review by maxpatiiuk — APPROVED** [review]
looks good - thanks for making the changes
see also my comment in https://github.com/specify/specify7/issues/4063#issuecomment-1871674519

**Review by pashiav — APPROVED** [review]
Renamed "Replace" to "Merge". Looks good!
issue-4063

**Comment by melton-jason** [comment]
> 1. I agree that it's redundant to have both update and delete
> 2. I see value in collection admins being able to allow someone to edit a record, without trusting them to bulk-edit those records (which is what record merging is essentially doing). Following this logic, to allow the user to merge Agents, they should have Agent delete, Agent update permissions and /record/replace permission. Is this what front-end and back-end are currently enforcing? if not, we should consider changing it

https://github.com/specify/specify7/issues/4063#issuecomment-1871674519

@specify/ux-testing, @specify/dev-testing 
I think these points are valid: the update and delete record merging permissions should be merged into one permission because they both are associated with the same action and are redundant with the Table update/delete permissions. 

Is this reasonable? 

Regardless, given that this PR already has three approvals, we can likely merge this into `xml-editor` and open a new PR if that is to be implemented.

**Pull request merged**